ipyvizzu.py

```diff
@@ -105,23 +109,40 @@ def __init__(self, **data):
     def feature(self, name, value):
         self._calls.append(Feature(name, value))
 
-    def animate(self, animation: typing.Optional[Animation]=None, **config):
+    def animate(self, *animations: typing.Optional[Animation], **options):
```

Comment:
The recursive merge of `_merge_anims` makes it hard to understand what kind of javascript will be generated. I would like to keep it simple. I think the `animate` method should allow to merge only different type of `Animation`s, so the merge would be only at the first level and only new keys would be allowed:

```python
class Animation:
    def dump(self):
        return json.dumps(self.build())

    @abc.abstractmethod
    def build(self) -> typing.Mapping:
        "Return a dict with native python values that can be converted into json"


class PlainAnimation(dict, Animation):                                                      
    def build(self):                                                        
        return self

                                                
class AnimationMerger(Animation):
    def __init__(self):
        self._data = {}

    def build(self):
        return self._data

    def merge(self, animation: Animation):
        data = self._validate_animation(animation)
        self._data.update(data)

    def _validate_animation(self, animation):
        data = animation.build()
        common_keys = set(data).intersection(self._data)
        assert not common_keys, f"Animation already set: {common_keys}"
        return data

       
class Chart:
    def animate(self, *animations: Animation, **options):
          merger = AnimationMerger()
          merger.megre(PlainAnimation(options))
          for animation in animations:
              merger.megre(animation)
          self._calls.append(Animate(merger))
```
In this way the `args` and `kwargs` can be merged, and both way looks similar:

```python
Chart().animate(config={...}, style={})
```

```python
Chart().animate(Config(...), Style(...))
```

Reply:
If you agree with this simplified merge strategy, I would merge the https://github.com/vizzuhq/ipyvizzu/pull/18.

Reply:
Thanks, I will think about it, yours seems more manageable.
My original idea was if I have more precreated Style Classes (for example one for the title and one for the plot),
it would be a use case to reuse their combinations parallel (if those do not modify the same style parameter).

Reply:
My original idea was to stay as close as possible to the original vizzu javascript API, so the documentation would be almost a copy/paste. That's why the user can pass plain dict, which is very similar to a javascript object (from syntax point of view). But I would like to avoid the typos therefore I added some helper classes with the fixed keys: `{"data": ...}` => `Data(...)`.

The lower level keys could follow a similar merge stategy on its own level. If we knew the available keys, then we can create a helper class for them:

```python
Style({
    "plot": {
        "marker": {
            "colorPalette": "#9355e8FF #123456FF #BDAF10FF"
        }
    }
})
```

```python
Style(
    Plot(
        Marker(
            ColorPalette("#9355e8FF #123456FF #BDAF10FF")
        )
    )
)
```

But we can be still flexible with the keyword arguments:

```python
Style(
    plot=...
)
```

Reply:
This is a clear solution. I like it.

Reply:
Sorry, I just saw your comment. Finish the discussion there.